pkg/gather/clusterconfig/csr.go

```diff
@@ -0,0 +1,185 @@
+package clusterconfig
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/openshift/insights-operator/pkg/utils"
+	"k8s.io/api/certificates/v1beta1"
+	certificatesv1b1api "k8s.io/api/certificates/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/json"
+)
+
+type CSRAnonymizer struct {
+	*certificatesv1b1api.CertificateSigningRequest
+}
+
+func (a CSRAnonymizer) Marshal(_ context.Context) ([]byte, error) {
```

Comment:
nitpick - all public methods should have proper docstring

Reply:
these comments are still pending, would be nice to add some docs in the whole file.